File: iPhone/VerBank-CSTSv3-IOSClientApi/ClientStation/api/info/InfoOperator_TRADESERV1005.m
//
//  InfoOperator_TRADESERV1005.m
//  VerBank-IOSClientStation
//
//  Created by Allone on 15/8/31.
//  Copyright (c) 2015年 zhanglei. All rights reserved.
//

#import "InfoOperator_TRADESERV1005.h"

#import "Info_TRADESERV1005.h"
#import "CDS_AccountStore.h"
#import "APIDoc.h"
#import "DocOperator.h"
#import "InstrumentDocOperator.h"
#import "UserDocOperator.h"

@implementation InfoOperator_TRADESERV1005

- (void)onInfo:(InfoFather *)info {
    Info_TRADESERV1005 *tinfo = (Info_TRADESERV1005 *)info;
    
    int attrID = [tinfo getChangedAttrID];
    switch (attrID) {
        case ATTRID_ACCOUNTSTRATEGY:
            for (CDS_AccountStore *as in [[APIDoc getUserDocCaptain] getCDS_AccountStoreArray]) {
                [[APIDoc getUserDocCaptain] killUserAccount:[as getAccountID]];
            }
            [[DocOperator getInstance] resetUserDoc];
            [APIDoc setInited:true];
            break;
            
        case ATTRID_INSTRUMENTCFG4ACCOUNT:
            if ([[APIDoc getUserDocCaptain] getCDS_AccountStore:[tinfo getAccountID]] == nil) {
                return;
            }
            
            if ([tinfo getInstrumentName] == nil) {
                [[InstrumentDocOperator getInstance] loadInstrumentsAccountCfg:[tinfo getAccountID]];
            } else {
                [[InstrumentDocOperator getInstance] loadInstrumentsAccountCfg:[tinfo getAccountID]
                                                                    instrument:[tinfo getInstrumentName]];
            }
            break;
            
        case ATTRID_INSTRUMENTCFG4GROUP:
            
            if ([[APIDoc getUserDocCaptain] getGroupDoc:[tinfo getGroupName]] == nil) {
                return;
            }
            
            if ([tinfo getInstrumentName] == nil) {
                [[InstrumentDocOperator getInstance] loadInstrumentsGroupCfg:[tinfo getGroupName]];
            } else {
                [[InstrumentDocOperator getInstance] loadInstrumentsGroupCfg:[tinfo getGroupName]
                                                                  instrument:[tinfo getInstrumentName]];
            }
            break;
            
        case ATTRID_USERLOGIN:
            if ([[APIDoc getUserDocCaptain] getUserLogin:[tinfo getAeid]] == nil) {
                return;
            }
            [[UserDocOperator getInstance] loadUserLogin:[tinfo getAeid]];
            break;
            
        case ATTRID_ACCOUNT_DELETE:
            [[APIDoc getUserDocCaptain] killUserAccount:[tinfo getAccountID]];
            break;
            
        default:
            break;
    }
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/client/station/api/info/InfoOperator_TRADESERV1005.java
package jedi.ex01.client.station.api.info;

import jedi.ex01.CSTS3.comm.info.InfoFather;
import jedi.ex01.CSTS3.comm.info.Info_TRADESERV1005;
import jedi.ex01.client.station.api.doc.DataDoc;
import jedi.ex01.client.station.api.doc.DocCaptain;
import jedi.ex01.client.station.api.doc.operator.DocOperator;
import jedi.ex01.client.station.api.doc.operator.instrument.InstrumentDocOperator;
import jedi.ex01.client.station.api.doc.operator.user.UserDocOperator;

public class InfoOperator_TRADESERV1005 extends InfoOperator {

	public void onInfo(InfoFather _info) {
		Info_TRADESERV1005 info = (Info_TRADESERV1005) _info;
		switch (info.getChangedAttrID()) {
			case Info_TRADESERV1005.ATTRID_ACCOUNTSTRATEGY:
				DocOperator.getInstance().resetUserDoc();
				DocCaptain.setInited(true);
				break;
			case Info_TRADESERV1005.ATTRID_INSTRUMENTCFG4ACCOUNT:
				if (DataDoc.getInstance().getAccountStore().getAccountID() != info.getAccountID()) {
					return;
				}
				if (info.getInstrumentName() == null) {
					InstrumentDocOperator.getInstrumentDocOperator().loadInstruments();
				} else {
					InstrumentDocOperator.getInstrumentDocOperator().loadInstruments(
							info.getInstrumentName());
				}
				break;
			case Info_TRADESERV1005.ATTRID_INSTRUMENTCFG4GROUP:
				if (!DataDoc.getInstance().getGroup().getGroupName().equals(info.getGroupName())) {
					return;
				}
				if (info.getInstrumentName() == null) {
					InstrumentDocOperator.getInstrumentDocOperator().loadInstruments();
				} else {
					InstrumentDocOperator.getInstrumentDocOperator().loadInstruments(
							info.getInstrumentName());
				}
				break;
			case Info_TRADESERV1005.ATTRID_USERLOGIN:
				if (!DataDoc.getInstance().getUserLogin().getAeid().equals(info.getAeid())) {
					return;
				}
				UserDocOperator.getUserDocOperator().loadUserLogin(info.getAeid());
				break;
			case Info_TRADESERV1005.ATTRID_ACCOUNT_DELETE:
				// DataDoc.getInstance().killUserAccount(new Long(tinfo.getAccountID()));
				break;
		}
	}
}
